### ISSUE TITLE
Allow custom ID field or fix HTML export

### ISSUE THREAD

**Issue by torfjelde** [org-roam/org-roam#883]
### Brief Abstract

I often export my notes to HTML, and org-roam actually provides the functionality for ensuring cross-references between files stay consistent. It would be awesome if references worked even after export!

### Proposed Implementation (if any)

To me it seems like the simplest solution is to allow the user to define the `ID` field, thus allowing them to override this to `CUSTOM_ID`. This will then leverage the existing functionality to ensure that ID's are referenced correctly in the exported files.

Currently you can make `org-id-link-to-org-use-id` to make org also respect the `ID` field but because the links created with using the `ID` field when doing `C-c C-l` results in a link of type `plain-text` since it does not contain the filename. This means that`org-html-link` doesn't treat the link as a `headline` and instead exports _links_ to this heading with a "ID-" concatenated to the ID field, thus breaking the link. So the _clean_ fix would be to ensure that `org-html-link` treats the `org-roam-store-link` as a headline. Suggestions?

(Awesome package btw! I have tons and tons of notes in org-mode and this seems absolutely perfect for improving the organization of them.)

**Comment by jethrokuan** [comment]
I looked into this briefly, it looks like prior to export, one can just `org-id-extra-flies` to include Org-roam files, and it should just work.

https://github.com/jethrokuan/braindump/blob/5cde89fc39b20c99278d6daaf9aea831bd7955e7/org/publish.el#L38

**Comment by sid-kurias** [comment]
I tried the above suggestion and still run into the issue with ox-html prepending ID- to the link.

Also org-roam does not seem to be using the CUSTOM_ID property if it exists. I have set org-id-link-to-use-id to 'create-if-interactive-and-no-custom-id. I guess that impacts store link only and not links that are created through org-roam completion. 

I am using doom emacs and here is the partial config
```` 
(after! org-roam
      (setq org-id-extra-files (org-roam--list-all-files)))
````

org-roam-version 1.2.3 and doom 2.0.9 and emacs 27.1 and org-mode 9.5
